ibc-core/ics02-client/types/src/error.rs

```diff
@@ -4,31 +4,34 @@ use core::convert::Infallible;
 
 use displaydoc::Display;
 use ibc_core_commitment_types::error::CommitmentError;
-use ibc_core_host_types::error::IdentifierError;
+use ibc_core_host_types::error::{DecodingError, IdentifierError};
 use ibc_core_host_types::identifiers::ClientId;
 use ibc_primitives::prelude::*;
 use ibc_primitives::Timestamp;
 
-use super::status::Status;
 use crate::height::Height;
+use crate::StatusError;
 
 /// Encodes all the possible client errors
 #[derive(Debug, Display)]
 pub enum ClientError {
     /// upgrade client error: `{0}`
     Upgrade(UpgradeClientError),
-    /// invalid client status: `{0}`
-    InvalidStatus(Status),
+    /// decoding error: `{0}`
+    Decoding(DecodingError),
+    /// client status error: `{0}`
+    ClientStatus(StatusError),
+    /// invalid header type: `{0}`
+    InvalidHeaderType(String),
     /// invalid trust threshold: `{numerator}`/`{denominator}`
     InvalidTrustThreshold { numerator: u64, denominator: u64 },
     /// invalid client state type: `{0}`
     InvalidClientStateType(String),
     /// invalid client consensus state type: `{0}`
     InvalidConsensusStateType(String),
-    /// invalid header type: `{0}`
-    InvalidHeaderType(String),
     /// invalid update client message
     InvalidUpdateClientMessage,
+    // TODO(seanchen1991): Should this be removed in favor of DecodingError::InvalidIdentifier?
     /// invalid client identifier: `{0}`
     InvalidClientIdentifier(IdentifierError),
```

Comment:
As chatted, variants like `InvalidClientIdentifier` can be merged into the `DecodingError` category. There are several other variants throughout our enum errors that occur during the decoding process, and they can also be consolidated. For example, variants like `InvalidRawHeader`, `MissingRawMisbehaviour`, and `MissingRawClientState` can all fall under `DecodingError`. This would further reduce the number of variants in the error enums.

Reply:
[f414ff7](https://github.com/cosmos/ibc-rs/pull/1325/commits/f414ff734fab0c169022079bc7940ceaed5fb810)

Reply:
Yes actually. There are bunch of other variants across the entire repo that can be consolidated in the same way.